core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -26,74 +34,97 @@
     }
 
     @SuppressWarnings("WeakerAccess")
-    public DockerComposeContainer(File composeFile, String command) {
-        super("dduportal/docker-compose:1.6.0");
-
-        // Create a unique identifier and tell compose
-        identifier = Base58.randomString(6).toLowerCase();
-        addEnv("COMPOSE_PROJECT_NAME", identifier);
+    public DockerComposeContainer(File composeFile, String identifier) {
+        this.composeFile = composeFile;
 
-        // Map the docker compose file into the container
-        addEnv("COMPOSE_FILE", "/compose/" + composeFile.getAbsoluteFile().getName());
-        addFileSystemBind(composeFile.getAbsoluteFile().getParentFile().getAbsolutePath(), "/compose", READ_ONLY);
+        // Use a unique identifier so that containers created for this compose environment can be identified
+        this.identifier = identifier;
 
-        // Ensure that compose can access docker. Since the container is assumed to be running on the same machine
-        //  as the docker daemon, just mapping the docker control socket is OK.
-        // As there seems to be a problem with mapping to the /var/run directory in certain environments (e.g. CircleCI)
-        //  we map the socket file outside of /var/run, as just /docker.sock
-        addFileSystemBind("/var/run/docker.sock", "/docker.sock", READ_WRITE);
-        addEnv("DOCKER_HOST", "unix:///docker.sock");
-
-        if (command != null) {
-            setCommand(command);
-        }
+        this.dockerClient = DockerClientFactory.instance().client();
     }
 
-    @Override
-    public void start() {
-
+    @Override @VisibleForTesting
+    public void starting(Description description) {
         final Profiler profiler = new Profiler("Docker compose container rule");
         profiler.setLogger(logger());
         profiler.start("Docker compose container startup");
 
+        applyScaling(); // scale before up, so that all scaled instances are available first for linking
+        createServices();
+        registerContainersForShutdown();
+        startAmbassadorContainers(profiler);
+
+    }
+
+
+    private void createServices() {
         // Start the docker-compose container, which starts up the services
-        super.start();
-        followOutput(new Slf4jLogConsumer(logger()), OutputFrame.OutputType.STDERR);
+        getDockerCompose("up -d")
+                .start();
+    }
 
-        // wait for the compose container to stop, which should only happen after it has spawned all the service containers
-        logger().info("Docker compose container is running - service creation will start now");
-        while (isRunning()) {
-            logger().trace("Compose container is still running");
-            Uninterruptibles.sleepUninterruptibly(100, TimeUnit.MILLISECONDS);
+    private DockerCompose getDockerCompose(String cmd) {
+        return new DockerCompose(composeFile, identifier)
+                .withCommand(cmd)
+                .withEnv(env);
```

Comment:
Docker-compose is launched with: 

* a filesystem bind that is the parent folder of the compose file
* the environment that has been set on `DockerComposeContainer`

This seems sufficient, unless there are usages that I'm not aware of yet!